### PR TITLE
refactore(add store)

### LINKED ISSUE #84: Update Pydantic to version 2

### PR DESCRIPTION
**Pull request by ChrOertlin**
### Added

adds a store to the genotype api.

Database is now initialised on startup and sessions are closed on app shutdown.

Also migrated to pydantic v2 in one go, i..e this closes #84, #94

### Fixed

**Steps to consider while deploying**
- Configuration changes:
- Documentation updates:
- Inform users by email:

### Review:
- [ ] Code approved by
- [ ] Tests executed on stage by:  (Document the test done with screen shots and description.)
- [ ] "Merge and deploy" approved by

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: genotype_api/services/endpoint_services/sample_service.py [hunk 1]
```diff
@@ -78,30 +60,35 @@ def _get_sample_response(self, sample: Sample) -> SampleResponse:
...
-        samples: list[Sample] = get_filtered_samples(
-            session=self.session, filter_params=filter_params
-        )
+        samples: list[Sample] = self.store.get_filtered_samples(filter_params=filter_params)
         return [self._get_sample_response(sample) for sample in samples]
 
-    def create_sample(self, sample: Sample) -> None:
-        create_sample(session=self.session, sample=sample)
+    def create_sample(self, sample: SampleCreate) -> None:
+        new_sample: Sample = Sample(
```
Comment (diitaz93) [inline genotype_api/services/endpoint_services/sample_service.py:75]: type hint redundant as preceding constructor

File: genotype_api/services/endpoint_services/sample_service.py [hunk 2]
```diff
@@ -78,30 +60,35 @@ def _get_sample_response(self, sample: Sample) -> SampleResponse:
...
     def get_samples(self, filter_params: SampleFilterParams) -> list[SampleResponse]:
-        samples: list[Sample] = get_filtered_samples(
-            session=self.session, filter_params=filter_params
-        )
+        samples: list[Sample] = self.store.get_filtered_samples(filter_params=filter_params)
         return [self._get_sample_response(sample) for sample in samples]
 
-    def create_sample(self, sample: Sample) -> None:
-        create_sample(session=self.session, sample=sample)
+    def create_sample(self, sample: SampleCreate) -> None:
```
Comment (diitaz93) [inline genotype_api/services/endpoint_services/sample_service.py:74]: The naming of this parameter is confusing, would you consider renaming it?
```suggestion
    def create_sample(self, sample_create: SampleCreate) -> None:
```

**Review by diitaz93 — APPROVED** [review]
Looks great 💯

**Comment by ChrOertlin** [comment]
# tests
Database connection via the swagger works - added a test user 
image

**Comment by ChrOertlin** [comment]
# plates view works:
image

**Comment by ChrOertlin** [comment]
# Specific plate view works:
image

**Comment by ChrOertlin** [comment]
# matches to work:
image

**Comment by ChrOertlin** [comment]
# read snps works
image

**Pull request merged**